code/game/objects/items/kitchen.dm

```diff
@@ -160,6 +160,21 @@
 	create_reagents(5, INJECTABLE|OPENCONTAINER|DUNKABLE)
 	register_item_context()
 
+/obj/item/kitchen/spoon/create_reagents(max_vol, flags)
+	. = ..()
+	RegisterSignals(reagents, list(COMSIG_REAGENTS_NEW_REAGENT, COMSIG_REAGENTS_ADD_REAGENT, COMSIG_REAGENTS_DEL_REAGENT, COMSIG_REAGENTS_REM_REAGENT), PROC_REF(on_reagent_change))
+	RegisterSignal(reagents, COMSIG_PARENT_QDELETING, PROC_REF(on_reagents_del))
```

Comment:
You don't need to do this, IIRC signals are already cleared by default on qdel

Reply:
https://github.com/tgstation/tgstation/blob/9f2356d30baa35428273096093ab261344e3878a/code/datums/datum.dm#L143 

Looks like it's being cleared properly, only exception I know of is turfs. You just found some easy code improvements/fixes